### PR TITLE
Support running Stratum-bfrt on SONiC-based OS

### PR DESCRIPTION
**Pull request by Yi-Tseng**
Use `/etc/sonic/sonic-environment` or ` /etc/sonic/config_db.json` file to load the PLATFORM string.

### CODE REVIEW

File: stratum/hal/bin/barefoot/docker/start-stratum-container.sh [hunk 1]
```diff
@@ -4,13 +4,18 @@
 set -e
 
 LOG_DIR=${LOG_DIR:-/var/log}
-SDE_VERSION=${SDE_VERSION:-9.5.0}
+SDE_VERSION=${SDE_VERSION:-9.7.0}
```
Comment (pudelkoM) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:7]: I don't think we should update the default version here/now. How about waiting for the flip in fabric-tna, then updating the default everywhere?

File: stratum/hal/bin/barefoot/docker/start-stratum-container.sh [hunk 2]
```diff
@@ -52,5 +57,6 @@ docker run -it --rm --privileged \
     $ONLP_MOUNT \
     $CHASSIS_CONFIG_MOUNT \
     -v $LOG_DIR:/var/log/stratum \
+    --name stratum \
```
Comment (pudelkoM) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:60]: Good idea, I'd like to apply this to every container

File: stratum/hal/bin/barefoot/docker/start-stratum-container.sh [hunk 3]
```diff
@@ -11,6 +11,11 @@ DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG:-$SDE_VERSION}
 # Try to load the platform string if not already set.
 if [[ -z "$PLATFORM" ]] && [[ -f "/etc/onl/platform" ]]; then
     PLATFORM=$(cat /etc/onl/platform)
+elif [[ -z "$PLATFORM" ]] && [[ -f "/etc/sonic/sonic-environment" ]]; then
+    source /etc/sonic/sonic-environment
+    PLATFORM=$(echo "$PLATFORM" | sed 's/_/-/g')
```
Comment (bocon13) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:16]: Do we care about "polluting" this script with other variables from `sonic-environment`?

Note: The variables are not exported to subprocesses unless you call `set -o allexport` before `source`. So this really only affects the code in this file.

If we do care, here's one solution:
```suggestion
    PLATFORM=$(source /etc/sonic/sonic-environment; echo $PLATFORM | sed 's/_/-/g')
```

Test:
```bash
$ echo $(source /etc/sonic/sonic-environment; echo $PLATFORM | sed 's/_/-/g')
x86-64-accton-wedge100bf-32x-r0
```

Comment (bocon13) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:16]: Do we care about "polluting" this script with other variables from `sonic-environment`?

Note: The variables are not exported to subprocesses unless you call `set -o allexport` before `source`. So this really only affects the code in this file.

If we do care, here's one solution:
```suggestion
    PLATFORM=$(source /etc/sonic/sonic-environment; echo "$PLATFORM" | sed 's/_/-/g')
```

Test:
```bash
$ echo $(source /etc/sonic/sonic-environment; echo $PLATFORM | sed 's/_/-/g')
x86-64-accton-wedge100bf-32x-r0
```

File: stratum/hal/bin/barefoot/docker/start-stratum-container.sh [hunk 4]
```diff
@@ -11,6 +11,11 @@ DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG:-$SDE_VERSION}
...
+    # Disable ONLP when running Stratum on SONiC.
+    set -- ${@} --enable_onlp=false
```
Comment (bocon13) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:18]: This is a little bit cryptic to me.

We might consider explicitly defining:
```
STRATUM_ARGS=${@}
```

Then, just the following here:
```
STRATUM_ARGS+=" --enable_onlp=false"
```

What do you think?

Comment (Yi-Tseng) [reply stratum/hal/bin/barefoot/docker/start-stratum-container.sh:18]: Sounds good to me

**Review by bocon13 — APPROVED** [review]
lgtm

**Review by pudelkoM — DISMISSED** [review]

**Review by pudelkoM — APPROVED** [review]
LGTM, feel free to merge once you finished your testing on SONiC base OS.

**Pull request closed without merging**